functions/v2/http-logging/main_test.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import flask
+import pytest
+
+import main
+
+
+# Create a fake "app" for generating test request contexts.
+@pytest.fixture(scope="module")
+def app():
+    return flask.Flask(__name__)
+
+
+def test_functions_log_http_should_print_message(app, capsys):
+    with app.test_request_context():
+        res = main.structured_logging(flask.request)
+        out, _ = capsys.readouterr()
+        assert "Hello, world!" in res
+        assert "Wrote logs to New-Structured-Log" in out
```

Comment:
The log should be a JSON string in stdout, let's make sure the message, components and project ID show up there

Reply:
+1, also test for the expected request data